test/test_script.py

```diff
@@ -943,3 +943,23 @@ def test_cmd_slice(self):
             self.parser.eval("$slice(abc; def,0)")
         with self.assertRaisesRegex(ScriptError, areg):
             self.parser.eval("$slice(abc; def),0,1,:,extra")
+
+    def test_cmd_datetime(self):
+        context = Metadata()
+        context["foo"] = "%Y%m"
+        # Tests with context
+        self.assertScriptResultEquals("$len($datetime(%foo%))", "6", context)
```

Comment:
I wonder if it would make sense to have an optional parameter to pass a defined date & time to `$datetime()` so it wouldn't use `now()`. We could then test the exact returned value (a string representing a known time) and get rid of those `$len()`.

@phw what do you think?

Reply:
Added something.  For now I limited the input to the '%Y-%m-%d %H:%M:%S' format.  I thought about using the override format if supplied but opted not.  Can change this if you think it would be more consistent that way.

Reply:
What I was thinking about is to pass a fixed timestamp and ensure it results in something consistent.

Reply:
I think I may be misunderstanding.  What I've added was the ability to set the date/time used in the function output as an optional input to the function.  I'm now thinking that you might have been looking for a way to pass a fixed time not to the function itself, but to the test somehow.  If that's it, then I need to go away and think about how that might be done.

Reply:
I think I understand now.  Have changed the testing to use a fixed date/time.  Is this what you were thinking?